inginious/frontend/plugins/notebooks_grader/api/notebook_grader_api.py

```diff
@@ -337,6 +339,19 @@ def API_POST(self): # pylint: disable=arguments-differ
                         result, 
                         submission_grade, 
                         newsub=True)
+                    
+                    #Add Sumbission Analytic
+                    analytics_manager = AnalyticsCollectionManagerSingleton.get_instance()
+                    services_manager = ServicesCollectionManagerSingleton.get_instance()
+                    service = {
+                        "key": "notebook_client_grader",
+                        "name": "Notebook (External Grading) Submissions"
+                    }
+                    session_id = self.user_manager.session_id()
+                    date = datetime.datetime.now()
+                    analytics_manager.add_visit(service, username, date, session_id, course_id)
+                    services_manager.add_service(service)
```

Comment:
Is this the proper order to execute these instructions? IMO we should register the service and then the visit.